### PR TITLE
Elastic: request be more graphql

### PR DESCRIPTION
**Pull request by felixleo22**
i put tests in this pull request

### CODE REVIEW

File: api/lib/client.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const { elasticsearch } = require('config');
 
 module.exports = new Client({
   node: {
-    url: new URL(`http://${elasticsearch.host}:${elasticsearch.port}`),
+    url: new URL(`${elasticsearch.host}:${elasticsearch.port}`),
```
Comment (nojhamster) [inline api/lib/client.js:7]: Confusing, the `host` should not include the `protocol`

File: api/lib/elastic.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+const client = require('./client');
+const { logger } = require('./logger');
+const unpaywallIndex = require('../index/unpaywall.json');
+const taskIndex = require('../index/task.json');
```
Comment (nojhamster) [inline api/lib/elastic.js:4]:
```suggestion
const unpaywallTemplate = require('../index/unpaywall.json');
const taskTemplate = require('../index/task.json');
```

File: api/lib/elastic.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      logger.error(`indices.create in createIndex: ${err}`);
+    }
+  }
+};
+
+const initalizeIndex = async () => {
+  const up = await pingElastic();
+  if (up) {
+    await createIndex('unpaywall', unpaywallIndex);
+    await createIndex('task', taskIndex);
```
Comment (nojhamster) [inline api/lib/elastic.js:51]:
```suggestion
    await createIndex('unpaywall', unpaywallTemplate);
    await createIndex('task', taskTemplate);
```

**Review by nojhamster — COMMENTED** [review]
Partial review so far, will do the rest later

File: api/routers/enrich.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  const { args } = req.query;
+  let attrs = [];
+  if (args) {
+    const enrichAttributesJSON = setEnrichAttributesJSON();
+    attrs = checkAttributesJSON(args, enrichAttributesJSON);
+    if (!attrs) {
+      return res.status(401).json({ message: 'args incorrect' });
+    }
+  }
+  const file = await enrichmentFileJSON(req, attrs);
```
Comment (nojhamster) [inline api/routers/enrich.js:25]: A`try/catch` may be required here, as express router does not handle promises.

File: api/routers/enrich.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  let attrs = [];
+  if (args) {
+    const enrichAttributesCSV = setEnrichAttributesCSV();
+    attrs = checkAttributesCSV(args, enrichAttributesCSV);
+    if (!attrs) {
+      return res.status(401).json({ message: 'args incorrect' });
+    }
+  }
+  if (!separator) separator = ',';
+  const file = await enrichmentFileCSV(req, attrs, separator);
```
Comment (nojhamster) [inline api/routers/enrich.js:47]: A `try/catch` may be required here, as express router does not handle promises.

File: api/app.js [hunk 6]
```diff
@@ -55,14 +64,23 @@ app.use('/graphql', cors(corsOptions), bodyParser.json(), (req, res) => {
...
+
 app.use(RouterOutFiles);
 app.use(RouterTask);
-app.use(RouterManageDatabase);
+app.use(RouterEnrich);
 
-app.get('/ping', (req, res) => {
-  res.status(200).json({ data: 'pong' });
+app.get('/test', async (req, res) => {
+  await deleteTmpFile();
```
[the remark this answers is not in the document]
Comment (felixleo22) [reply api/app.js:76]: ping pong is used for uptime robot

File: api/lib/file.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+const fs = require('fs-extra');
+const path = require('path');
+const { logger } = require('./logger');
+
+const tmpDir = path.resolve(__dirname, '..', 'out', 'tmp');
+
+const deleteTmpFile = async () => {
+  const files = await fs.readdir(tmpDir);
+  files.forEach(async (file) => {
```
[the remark this answers is not in the document]
Comment (felixleo22) [reply api/lib/file.js:9]: can i use a "classic for" to do it ?

**Pull request merged**